source/material_model/visco_plastic.cc

```diff
@@ -191,15 +197,34 @@ namespace aspect
           // Convert friction angle from degrees to radians
           double phi = angles_internal_friction[j] * numbers::PI/180.0;
 
+          // Assing cohesions to a new variable
+          double coh = cohesions[j];
+
+          // Strain weakening
+          double strain_ii = 0;
+          if (use_strain_weakening == true)
+
+            // Calculate second strain invariant
+            strain_ii = ( (dim==3)
```

Comment:
In order to use those function you will need to convert the vector of doubles to a symmetric tensor. This is for example done in source/particle/property/integrated_strain.cc:48. Always best to copy existing code, that minimizes room for errors (and we had a lot of trouble with second invariants of strain and stress in the past).